cmdx.py

```diff
@@ -2555,20 +2604,20 @@ def type(self):
 
         return self._mplug.attribute().apiTypeStr
 
-    def path(self):
-        return "%s.%s" % (
-            self._node.path(), self._mplug.partialName(
+    def path(self, full=True):
```

Comment:
Interesting. Does this solve the issue of overly verbose paths e.g. `myNode.translate.translateX`? Or what was the motivation here?

Reply:
that's it. but it was part of my first findings when i began to refactor existing code. maybe we should have more options to translate plug names like cmds (like shortnames)

i also wondered why you were using full path and not shortest path when you convert node and plugs to string. is it for performance issues? if not it would be better to use shortest path to find back the original maya behaviour to print node names. but again i'm not really sure of the initial intention of it

Reply:
I've found that paths are rarely useful with cmdx, since nodes are uniquely identified via their hash. And when I do want to know the path, I found it useful to mimic what filesystems do.

```c
c:\my\full\path

# Versus
full\path
```

Especially since the shortest path in Maya changes based on what else is around. It reads to me like an optimisation for serialisation (like the shortName of attributes) rather than anything someone would want to read.

In this case, `.translate.translateX` is *not* what one would expect to read, so I think we're moving in the right direction.